hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -363,6 +363,12 @@ class EmitFunctionBuilder[F >: Null](
     _hfield.load()
   }
 
+  val lazyFieldMemo: mutable.Map[Any, LazyFieldRef[_]] = mutable.Map.empty
+
+  def getOrDefineLazyField[T: TypeInfo](setup: Code[T], id: Any): LazyFieldRef[T] = {
```

Comment:
Is there any reason this shouldn't be on `FunctionBuilder`?

Reply:
er, that would be fine too.